compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUCheckIRBeforeLLVMConversion.cpp

```diff
@@ -58,11 +59,14 @@ checkStackAllocationSize(mlir::FunctionOpInterface funcOp) {
       allocaSize *= dimSize;
     }
     for (auto operand : allocaOp.getDynamicSizes()) {
-      auto ub = ValueBoundsConstraintSet::computeConstantBound(
-          presburger::BoundType::UB, operand, /*dim=*/std::nullopt,
-          /*stopCondition=*/nullptr, /*closedUB=*/true);
-      if (succeeded(ub)) {
-        allocaSize *= ub.value();
+      // Assume vscale is 1 for determining if the alloca is within the stack
+      // limit. This should always resolve to a constant bound.
```

Comment:
Ah, now I see, thanks for the explanation!

Would it be possible to grab the actual values though? I guess that it can only come from a function attribute and/or target info? Otherwise I'm worried that this isn't conservative enough 🤔

Reply:
As far as I understand it this is just a check for unreasonably large allocations, but nothing would break if at some `vscale` it happened to go over it a bit. In the size of allocations `vscale` will a small factor (with the most of the size likely from static dimensions), so I think you can get away with mostly ignoring it (and still have a reasonable check).

Reply:
Assuming `vscale = 16` (on Arm) would be assuming a value of vscale that is very unlikely to actually occur.

Reply:
Yes, `vscale = 16` would be super conservative. AFAIK, the widest vectors that we have today are 512 bits (`vscale = 4`):
* https://en.wikipedia.org/wiki/Fujitsu_A64FX

However, we need to make sure that we don't underestimate too optimistically. It's tricky because this is a compile-time check and we won't know the value of `vscale` until runtime. I think that `clAssumedVscaleValue` that you added is a very good compromise for now. If we need to refine this logic then what you've added here will allow us to make an informed decision. Thanks!